examples/squeezer_bin/squeezer_bin/imp.rs

```diff
@@ -0,0 +1,106 @@
+use gtk::glib;
+use gtk::glib::{ParamSpec, Value};
+use gtk::gsk;
+use gtk::prelude::*;
+use gtk::subclass::prelude::*;
+
+use once_cell::sync::Lazy;
+use std::cell::Cell;
+
+#[derive(Debug)]
+pub struct SqueezerBin {
+    pub keep_aspect_ratio: Cell<bool>,
+}
+
+#[glib::object_subclass]
+impl ObjectSubclass for SqueezerBin {
+    const NAME: &'static str = "SqueezerBin";
+    type ParentType = gtk::Widget;
+    type Type = super::SqueezerBin;
+
+    fn new() -> Self {
+        Self {
+            keep_aspect_ratio: Cell::new(false),
+        }
+    }
+}
+
+impl ObjectImpl for SqueezerBin {
+    fn properties() -> &'static [glib::ParamSpec] {
+        static PROPERTIES: Lazy<Vec<glib::ParamSpec>> = Lazy::new(|| {
+            vec![glib::ParamSpecBoolean::new(
+                "keep-aspect-ratio",
+                "Keep aspect ratio",
+                "If true it keeps the aspect ratio of the widget",
+                false,
+                glib::ParamFlags::READWRITE | glib::ParamFlags::EXPLICIT_NOTIFY,
+            )]
+        });
+
+        PROPERTIES.as_ref()
+    }
+
+    fn property(&self, obj: &Self::Type, _id: usize, pspec: &ParamSpec) -> Value {
+        match pspec.name() {
+            "keep-aspect-ratio" => obj.keep_aspect_ratio().to_value(),
+            _ => unimplemented!(),
+        }
+    }
+
+    fn set_property(&self, obj: &Self::Type, _id: usize, value: &Value, pspec: &ParamSpec) {
+        match pspec.name() {
+            "keep-aspect-ratio" => {
+                obj.set_keep_aspect_ratio(value.get().unwrap());
+            }
+            _ => unimplemented!(),
+        }
+    }
+
+    fn constructed(&self, obj: &Self::Type) {
+        obj.set_halign(gtk::Align::Fill);
+        obj.set_valign(gtk::Align::Fill);
+        obj.set_hexpand(true);
+        obj.set_vexpand(true);
+
+        let child = gtk::Label::new(Some("Hello World!"));
```

Comment:
It would seem nicer API-wise to not have a child built-in but to have API on the bin that allows the application to set a child. Similar to https://github.com/gtk-rs/gtk4-rs/pull/740